ehrapy/api/preprocessing/encoding/_encode.py

```diff
@@ -496,6 +597,92 @@ def _undo_encoding(
             layers={"original": new_x.copy()},
         )
 
+    @staticmethod
+    def _delete_all_encodings(adata) -> Tuple[Optional[np.ndarray], Optional[list]]:
+        """Delete all encoded columns and keep track of their indices
+        Args:
+            adata: The AnnData object to operate on
+
+        Returns:
+            A temporary X were all encoded columns are deleted and all var_names of unencoded columns
+        """
+        var_names = list(adata.var_names)
+        if adata.X is not None and var_names is not None:
+            idx = 0
+            for var_name in var_names:
+                if not var_name.startswith("ehrapycat"):
+                    break
+                idx += 1
+            # case: only encoded columns were found
+            if idx == len(var_names):
+                return None, None
+            # don't need to consider case when no encoded columns are there, since undo_encoding would not run anyways in this case
+            return adata.X[:, idx:].copy(), var_names[idx:]
+
+    @staticmethod
+    def _reorder_encodings(adata: AnnData, new_encodings):
+        """Reorder the encodings and update which column will be encoded using which mode (with which columns in case of
+        multi column encoding modes).
+        Args:
+            adata: The AnnData object to be reencoded
+            new_encodings: The new encodings passed by the user (might affect encoded as well as previously non encoded columns)
+
+        Returns:
+            An updated encoding scheme
+        """
+        flattened_modes = sum(new_encodings.values(), [])
+        latest_encoded_columns = list(chain(*(i if isinstance(i, list) else (i,) for i in flattened_modes)))
+        # check for duplicates and raise an error if any
+        if len(set(latest_encoded_columns)) != len(latest_encoded_columns):
+            print(
+                "[bold red]Reencoding AnnData object failed. You have at least one duplicate in your encodings. A column "
+                "cannot be encoded at the same time using different encoding modes!"
+            )
+            raise DuplicateColumnEncodingError
+        old_encode_mode = adata.uns["current_encodings"]
+        for categorical in latest_encoded_columns:
+            encode_mode = old_encode_mode.get(categorical)
+            # if None, this categorical has not been encoded before but will be encoded now for the first time
+            # multi column encoding mode
+            if encode_mode in Encoder.multi_encoding_modes:
+                encoded_categoricals_with_mode = adata.uns["categoricals_encoded_with_mode"][encode_mode]
+                _found = False
+                for column_list in encoded_categoricals_with_mode:
+                    for column_name in column_list:
+                        if column_name == categorical:
+                            column_list.remove(column_name)
+                            _found = True
```

Comment:
Maybe there's a better way than defining a flag just to break out of a double for loop.